### PR TITLE
Changed_services

### PR DESCRIPTION
**Pull request by Kostya-Vystorobets**

### CODE REVIEW

File: controllers/employee.js [hunk 1]
```diff
@@ -2,6 +2,7 @@ const service = require('../services/employee')
 
 const getById = async (request, response, next) => {
     try {
+        console.log(request)
```
Comment (didevgen) [inline controllers/employee.js]: remove
**BUT** set up the [Winston](https://github.com/winstonjs/winston) logger

File: controllers/employee.js [hunk 2]
```diff
@@ -10,15 +11,15 @@ const getById = async (request, response, next) => {
 }
 const deleteById = async (request, response, next) => {
     try {
-        const employee = await service.deleteById({ _id: request.params.id })
+        const employee = await service.deleteById(request.params.departmentId, request.params.id)
         return response.send(employee)
```
Comment (didevgen) [inline controllers/employee.js]: don't think this employee should be returned if it has been deleted

File: routes/employee.js [hunk 3]
```diff
@@ -3,8 +3,8 @@ const controller = require('../controllers/employee')
 const router = express.Router()
 
 router.get('/:id', controller.getById)
-router.delete('/:id', controller.deleteById)
-router.post('/department/id', controller.create)
+router.delete('/:id/department/:departmentId', controller.deleteById)
```
Comment (didevgen) [inline routes/employee.js]: you need both options:
delete employee by employee id 
and delete employee from department

File: .prettierrc [hunk 4]
```diff
@@ -1,6 +0,0 @@
-{
-    "trailingComma": "es5",
-    "tabWidth": 4,
-    "semi": false,
-    "singleQuote": true
-}
```
Comment (didevgen) [inline .prettierrc:6]: Why did it get removed?

File: app.js [hunk 5]
```diff
@@ -8,13 +8,28 @@ const employeeRoutes = require('./routes/employee');
...
+//     try {
+//         await mongoose.connect(process.env.mongoURI);
+//         console.log("Connected to MongoDB.");
+//         await checkAnyUserExist();
+//     } catch (error) {
+//         console.error(error);
+//     }
+// };
+
+// initDatabase();
```
Comment (didevgen) [inline app.js]: delete

File: middleware/checkAnyUserExist.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    try {
+        const userCount = await User.find().count();
+        if (userCount === 0) {
+            const result = await createDefaultUser();
+            console.log(`username: ${result.username}`)
+            console.log(`password: ${result.password}`)
+        }
+    } catch (error) {
+        next(error);
+    }
```
Comment (didevgen) [inline middleware/checkAnyUserExist.js]: https://dev.to/dev_tycodez/seed-mongodb-mongoose-seed-51d5

File: middleware/verifyAuth.js [hunk 7]
```diff
@@ -0,0 +1,26 @@
+const jwt = require('jsonwebtoken');
+const ApplicationError = require('../errors/aplicationError');
+const logger = require("../logger/dev-logger");
+
+const verifyAuth = (request, response, next) => {
+    try {
+        if (request.path === "/api/v1/user/login") {
+            next();
+        } else {
+            const token = require.header("x-auth-token")
```
Comment (didevgen) [inline middleware/verifyAuth.js]: probably `request.header`

File: services/user.js [hunk 8]
```diff
@@ -1,31 +1,81 @@
...
+    if (user) {
+        const validPassword = await bcrypt.compare(request.body.password, user.password)
+        if (validPassword) {
+            const token = jwt.sign({
+                email: user.email,
+                userId: user._id
+            }, 'dev-jwt', { expiresIn: '24h' })
+            return token
+        } else {
+            throw new ApplicationError("Password is not correct", 401)
```
Comment (didevgen) [inline services/user.js]: user not found 404

File: services/user.js [hunk 9]
```diff
@@ -1,31 +1,81 @@
...
+        }
+    } else {
+        throw new ApplicationError("User not found", 401)
+    }
 };
 
-const logout = (request, response) => {
+const create = async (user) => {
+    console.log("work:createUser", user)
```
Comment (didevgen) [inline services/user.js]: you've setup logger

File: services/user.js [hunk 10]
```diff
@@ -1,31 +1,81 @@
...
+//         } else {
+//             response.status(401).json({ message: "Password is not correct" })
+//         }
+//     } else {
+//         response.status(404).json({ message: "User not found" })
+//     }
+// };
+
+// const logout = (request, response) => {
+
```
Comment (didevgen) [inline services/user.js]: remove

File: .prettierrc (hunk 4 above)
Comment (Kostya-Vystorobets) [reply .prettierrc:6]: I don't like the work of my prettier. For example, the service/employee.js file is not fixed, I could not understand why.  I wanted to reconfigure prettier and configure it correctly.

File: middleware/checkAnyUserExist.js (hunk 6 above)
Comment (Kostya-Vystorobets) [reply middleware/checkAnyUserExist.js]: Please tell me what exactly I need to do

File: combined.log [hunk 11]
```diff
@@ -0,0 +1,98 @@
...
+2022-04-11 10:12:07 [32minfo[39m: Connected to MongoDB.
+2022-04-11 10:23:28 [32minfo[39m: Server has been started on 3000
+2022-04-11 10:23:29 [32minfo[39m: Connected to MongoDB.
+2022-04-11 10:25:16 [32minfo[39m: Server has been started on 3000
+2022-04-11 10:25:17 [32minfo[39m: Connected to MongoDB.
+2022-04-11 10:26:15 [32minfo[39m: Server has been started on 3000
+2022-04-11 10:26:16 [32minfo[39m: Connected to MongoDB.
+2022-04-11 10:28:23 [32minfo[39m: Server has been started on 3000
+2022-04-11 10:28:24 [32minfo[39m: Connected to MongoDB.
+2022-04-11 10:29:22 [32minfo[39m: Server has been started on 3000
```
Comment (didevgen) [inline combined.log]: should not be committed

File: data/users/users.js [hunk 12]
```diff
@@ -0,0 +1,6 @@
+module.exports = {
+    username: 'Admin',
+    password: 'PTt2KMj233s',
```
Comment (didevgen) [inline data/users/users.js]: :/

Comment (didevgen) [reply data/users/users.js]: should be provided from env vars

Comment (didevgen) [reply data/users/users.js]: and in hash

Comment (didevgen) [reply data/users/users.js]: or generate a random password in runtime

File: services/department.js [hunk 13]
```diff
@@ -1,26 +1,34 @@
...
 }
 
 const getById = async (departmentId) => {
-    const departmentById = Department.findById(departmentId)
-    return departmentById.populate("employees")
-
+    await validate.validateId(departmentId);
+    const department = Department.findById(departmentId);
+    if (!department) {
+        throw new ApplicationError('The department with this ID was not found', 400)
```
Comment (didevgen) [inline services/department.js:12]: 404

Comment (didevgen) [reply services/department.js:12]: and further occurances

File: validations/department.js [hunk 14]
```diff
@@ -0,0 +1,29 @@
+const Joi = require('joi');
+const ApplicationError = require('../errors/applicationError');
+
+const validDepartmentCreate = async (data) => {
+    try {
+        const schema = Joi.object({
+            name: Joi.string().alphanum().min(5).max(30).required(),
```
Comment (didevgen) [inline validations/department.js:7]: alphanumeric?

**Review by didevgen — APPROVED** [review]

**Pull request merged**